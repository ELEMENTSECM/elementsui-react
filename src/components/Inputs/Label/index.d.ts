import * as React from 'react';

export interface LabelProps {
        /**
         * Label required
         */
        required?: boolean;
        /**
         * Label disabled
         */
        disabled?: boolean;
        /**
         * Label  label
         */
        label: string;
        /**
         * User-defined styling
         */
        getStyles?: (...args: any[])=>any;
    }

    declare const Label: React.SFC<LabelProps>;
export default Label;