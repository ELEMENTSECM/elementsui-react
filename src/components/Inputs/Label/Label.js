import * as React from 'react';
import PropTypes from 'prop-types';
import { Label as UIFabLabel } from 'office-ui-fabric-react/lib/Label';

/** Label example */
function Label({ label, required, disabled }) {
	return (
		<UIFabLabel required={required} disabled={disabled}>
			{label}
		</UIFabLabel>
	);
}

Label.propTypes = {
	/**Label required */
	required: PropTypes.bool,
	/** Label disabled */
	disabled: PropTypes.bool,
	/** Label  label*/
	label: PropTypes.string.isRequired
};

export default Label;
