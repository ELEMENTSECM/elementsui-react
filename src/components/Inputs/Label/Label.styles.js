export const styles = props => {
	const { className, theme } = props;
	const { palette, semanticColors } = theme;

	return {
		root: [
			'ms-Label',
			{
				// place your styles here
			},
			className
		]
	};
};
