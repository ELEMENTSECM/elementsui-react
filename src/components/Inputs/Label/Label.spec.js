import React from 'react';
import renderer from 'react-test-renderer';
import Label from './Label';

describe('Label', () => {
	test('displays `Foo` as label', () => {
		const tree = renderer
			.create(
				<Label id="defaultLabel" required={false} disabled={false}>
					Foo
				</Label>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
