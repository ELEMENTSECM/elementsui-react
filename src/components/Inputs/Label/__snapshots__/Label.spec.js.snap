// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label displays \`Foo\` as label 1`] = `
<label
  className="ms-Label root-41"
  required={false}
>
  Foo
</label>
`;
